pkg/ui/src/util/locations.ts

```diff
@@ -89,3 +89,44 @@ export function findOrCalculateLocation(locations: LocationTree, locality: Local
   centroid = vector.mult(centroid, 1 / childLocations.length);
   return { longitude: centroid[0], latitude: centroid[1] };
 }
+
+/*
+ * getConfigStats finds how many nodes are missing locality or location config.
+ */
+export function getConfigStatus(
+  localityTree: LocalityTree, locationTree: LocationTree,
+): NodeMapConfigStatus {
+  let totalNodes = 0;
+  let withLocality = 0;
+  let withLocation = 0;
+  function recur(tree: LocalityTree) {
+    tree.nodes.forEach(() => {
+      totalNodes++;
+      // TODO: what about root?
+      if (tree.tiers.length > 0) {
+        withLocality++;
+        const currentTier = tree.tiers[tree.tiers.length - 1];
+        if (hasLocation(locationTree, currentTier)) {
+          withLocation++;
```

Comment:
yes, as you TODO'd above, localities have locations, not nodes.  it's not exactly clear how to specify "how many localities are left" in the general case.  perhaps rather than trying to be this specific, we should provide more general feedback and a link to the debug page?

Reply:
Done.